File: dev-docs/snippets/invitations/quota.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Check the caller's invitation quota. Admins/owners always get `unlimited: true`.
export async function quota(client: JsBaoClient) {
  // #region example
  const quota = await client.invitations.quota();
  const canInvite = quota.unlimited || quota.remaining > 0;
  // #endregion example
  return canInvite;
}

File: dev-docs/snippets/invitations/quota.swift
import JsBaoClient

// Check the caller's invitation quota. Returns a typed `InvitationQuota`.
func quota(client: JsBaoClient) async throws {
  // #region example
  let quota = try await client.invitations.quota()
  let canInvite = quota.unlimited || quota.remaining > 0
  // #endregion example
  _ = canInvite
}
